classes/ocis_manager.php

```diff
@@ -0,0 +1,496 @@
+<?php
+// This file is part of the ocis repository for Moodle - http://moodle.org/
+//
+// The ocis repository for Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The ocis repository for Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with the ocis repository for Moodle.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+namespace repository_ocis;
+
+use DateTime;
+use moodle_exception;
+use core\oauth2\issuer as oauth2_issuer;
+use core\oauth2\client as oauth2_client;
+use Owncloud\OcisPhpSdk\Drive;
+use Owncloud\OcisPhpSdk\DriveOrder;
+use Owncloud\OcisPhpSdk\DriveType;
+use Owncloud\OcisPhpSdk\Exception\BadRequestException;
+use Owncloud\OcisPhpSdk\Exception\ForbiddenException;
+use Owncloud\OcisPhpSdk\Exception\HttpException;
+use Owncloud\OcisPhpSdk\Exception\InternalServerErrorException;
+use Owncloud\OcisPhpSdk\Exception\InvalidResponseException;
+use Owncloud\OcisPhpSdk\Exception\NotFoundException;
+use Owncloud\OcisPhpSdk\Exception\UnauthorizedException;
+use Owncloud\OcisPhpSdk\Ocis;
+use Owncloud\OcisPhpSdk\OcisResource;
+use Owncloud\OcisPhpSdk\OrderDirection;
+
+/**
+ * Provides functions for managing access to the oCIS server.
+ *
+ * @package    repository_ocis
+ * @copyright  2023 ownCloud GmbH
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class ocis_manager {
+    private oauth2_client $oauth2client;
+    private oauth2_issuer $oauth2issuer;
+    private ?string $driveidandpath = null;
+    private bool $showpersonaldrive;
+    private bool $showshares;
+    private bool $showprojectdrives;
+    private ?string $driveid = null;
+    private string $path;
+    private ?Drive $drive = null;
+
+    public function __construct(
+        oauth2_client $oauth2client,
+        oauth2_issuer $oauth2issuer,
+        bool $showpersonaldrive,
+        bool $showshares,
+        bool $showprojectdrives
+    ) {
+        $this->oauth2client = $oauth2client;
+        $this->oauth2issuer = $oauth2issuer;
+        $this->showpersonaldrive = $showpersonaldrive;
+        $this->showshares = $showshares;
+        $this->showprojectdrives = $showprojectdrives;
+    }
+
+    private function get_drive_id(): string {
+        if ($this->driveid === null) {
+            $this->get_drive();
+        }
+        return $this->driveid;
+    }
+    private function get_drive(): Drive {
+        if ($this->drive !== null) {
+            return $this->drive;
+        } else {
+            // The colon ":" is the seperator between drive_id and path.
```

Comment:
```suggestion
            // The colon ":" is the separator between drive_id and path.
```

May as well fix this comment typo.